param/__init__.py

```diff
@@ -998,8 +998,15 @@ class Boolean(Parameter):
     # no effect since values are either False, True, or None (if allowed).
     def __init__(self, default=False, bounds=(0,1), **params):
         self.bounds = bounds
+        self._validate_default(default)
         super(Boolean, self).__init__(default=default, **params)
 
+    def _validate_default(self, default):
+        if not isinstance(default, bool) and default is not None:
+            raise ValueError("Boolean parameter only takes a "
+                             "Boolean default or None, not %s."
+                             % default)
+
```

Comment:
That logic looks good, thanks! I think we should follow the example of other Parameters like String and just call _validate in the constructor:

```
        self._validate(default)
```

Reply:
Thanks for the feedback :) I had a go at following what is done in Tuple in the latest commit. The only issue was having to modify the error message since `self.name` is not set when the validation happens and the message you would get, `"Boolean parameter None must be True or False, not {val}."`, seems confusing to me.